src/Modal/Modal.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { ReactNode, FC } from 'react'
+import styled, { css } from 'styled-components'
+import { theme } from '../theme'
+
+import { Box } from '../Box'
+import { Icon } from '../Icon'
+import { Text } from '../Text'
+
+interface IModalWrapper {
+  // showModal state
+  showModal: boolean
+}
+interface IModalContainer {
+  // drawer state
+  drawer: boolean
+}
+
+type ModalProps = {
+  title?: string,
+  children?: ReactNode,
```

Comment:
Should children be optional? It's optional here but has no default state when destructured in the component?